util/linuxfw/fakes.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build linux
+
+package linuxfw
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/google/nftables"
+	"github.com/mdlayher/netlink"
+	"github.com/vishvananda/netns"
+)
+
+var errExec = errors.New("execution failed")
+
+type fakeIPTables struct {
+	t *testing.T
+	n map[string][]string
+}
+
+type fakeRule struct {
+	table, chain string
+	args         []string
+}
+
+func NewIPTables(t *testing.T) *fakeIPTables {
```

Comment:
this needs to be its own package or something. otherwise we end up importing `testing` in production code.